<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>735</width>
    <height>672</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>648</width>
    <height>547</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>-1</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QScrollArea" name="networkArea">
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAsNeeded</enum>
        </property>
        <property name="widgetResizable">
         <bool>false</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>447</width>
           <height>552</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QWidget" name="verticalWidget" native="true">
      <property name="maximumSize">
       <size>
        <width>258</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="sizeConstraint">
        <enum>QLayout::SetMaximumSize</enum>
       </property>
       <item>
        <widget class="QTreeWidget" name="treeWidgetVirtualLinks">
         <property name="maximumSize">
          <size>
           <width>256</width>
           <height>16777215</height>
          </size>
         </property>
         <column>
          <property name="text">
           <string>Virtual links</string>
          </property>
         </column>
        </widget>
       </item>
       <item>
        <widget class="QTreeWidget" name="treeWidgetDataFlows">
         <property name="maximumSize">
          <size>
           <width>256</width>
           <height>16777215</height>
          </size>
         </property>
         <column>
          <property name="text">
           <string>Data flows</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>735</width>
     <height>32</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionSelect"/>
   <addaction name="actionEndSystem"/>
   <addaction name="actionPartition"/>
   <addaction name="actionSwitch"/>
   <addaction name="actionLink"/>
   <addaction name="separator"/>
   <addaction name="actionVirtualLink"/>
   <addaction name="actionRemoveVirtualLink"/>
   <addaction name="separator"/>
   <addaction name="actionDataFlow"/>
   <addaction name="actionRemoveDataFlow"/>
  </widget>
  <widget class="QToolBar" name="toolBar_2">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>true</bool>
   </attribute>
   <addaction name="actionVerify"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/folder.png</normaloff>:/pics/pics/folder.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/cd.png</normaloff>:/pics/pics/cd.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/page.png</normaloff>:/pics/pics/page.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionEndSystem">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/end_system_selected.png</normaloff>:/pics/pics/end_system_selected.png</iconset>
   </property>
   <property name="text">
    <string>EndSystem</string>
   </property>
   <property name="toolTip">
    <string>Add end system</string>
   </property>
  </action>
  <action name="actionSelect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/select.png</normaloff>:/pics/pics/select.png</iconset>
   </property>
   <property name="text">
    <string>Select</string>
   </property>
   <property name="toolTip">
    <string>Select elements of network</string>
   </property>
  </action>
  <action name="actionPartition">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/computer.png</normaloff>:/pics/pics/computer.png</iconset>
   </property>
   <property name="text">
    <string>Partition</string>
   </property>
   <property name="toolTip">
    <string>Add new partition</string>
   </property>
  </action>
  <action name="actionSwitch">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/router.png</normaloff>:/pics/pics/router.png</iconset>
   </property>
   <property name="text">
    <string>Switch</string>
   </property>
   <property name="toolTip">
    <string>Add AFDX switch</string>
   </property>
  </action>
  <action name="actionLink">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/edge.png</normaloff>:/pics/pics/edge.png</iconset>
   </property>
   <property name="text">
    <string>link</string>
   </property>
   <property name="toolTip">
    <string>Add link</string>
   </property>
  </action>
  <action name="actionVirtualLink">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/topology.png</normaloff>:/pics/pics/topology.png</iconset>
   </property>
   <property name="text">
    <string>virtualLink</string>
   </property>
   <property name="toolTip">
    <string>Add new virtual link</string>
   </property>
  </action>
  <action name="actionRemoveVirtualLink">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/delete.png</normaloff>:/pics/pics/delete.png</iconset>
   </property>
   <property name="text">
    <string>removeVirtualLink</string>
   </property>
   <property name="toolTip">
    <string>Remove virtual link</string>
   </property>
  </action>
  <action name="actionDataFlow">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/add.png</normaloff>:/pics/pics/add.png</iconset>
   </property>
   <property name="text">
    <string>dataFlow</string>
   </property>
   <property name="toolTip">
    <string>Add new data flow</string>
   </property>
  </action>
  <action name="actionRemoveDataFlow">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/pics/pics/delete.png</normaloff>:/pics/pics/delete.png</iconset>
   </property>
   <property name="text">
    <string>removeDataFlow</string>
   </property>
   <property name="toolTip">
    <string>Remove data flow</string>
   </property>
  </action>
  <action name="actionVerify">
   <property name="text">
    <string>verify</string>
   </property>
   <property name="toolTip">
    <string>Verify capacity constraints</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionEndSystem</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleEndSystem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSelect</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleSelect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>New()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Load()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPartition</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>togglePartition()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSwitch</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleSwitch()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionVirtualLink</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleVirtualLink()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemoveVirtualLink</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleRemoveVirtualLink()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDataFlow</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleDataFlow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemoveDataFlow</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleRemoveDataFlow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLink</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleLink()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidgetVirtualLinks</sender>
   <signal>itemClicked(QTreeWidgetItem*,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleVirtualLinkItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>596</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidgetVirtualLinks</sender>
   <signal>itemDoubleClicked(QTreeWidgetItem*,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleVirtualLinkDoubleClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>596</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidgetDataFlows</sender>
   <signal>itemClicked(QTreeWidgetItem*,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleDataFlowItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>596</x>
     <y>493</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidgetDataFlows</sender>
   <signal>itemDoubleClicked(QTreeWidgetItem*,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleDataFlowDoubleClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>596</x>
     <y>493</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionVerify</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>verify()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>toggleEndSystem()</slot>
  <slot>toggleSelect()</slot>
  <slot>Save()</slot>
  <slot>Load()</slot>
  <slot>New()</slot>
  <slot>togglePartition()</slot>
  <slot>toggleSwitch()</slot>
  <slot>toggleLink()</slot>
  <slot>toggleVirtualLink()</slot>
  <slot>toggleRemoveVirtualLink()</slot>
  <slot>toggleDataFlow()</slot>
  <slot>toggleRemoveDataFlow()</slot>
  <slot>toggleVirtualLinkItem()</slot>
  <slot>toggleVirtualLinkDoubleClicked()</slot>
  <slot>toggleDataFlowItem()</slot>
  <slot>toggleDataFlowDoubleClicked()</slot>
  <slot>verify()</slot>
 </slots>
</ui>
